backend/tests/api/users/test_register.py

```diff
@@ -20,29 +20,51 @@ def test_register_user(client, session):
 
     expected_response_2 = {"id": 2, "username": "John", "email": None}
 
-    response = client.post("users", data=registration_form_1)
+    response = client.post("/users", data=registration_form_1)
 
     assert response.status_code == status.HTTP_201_CREATED
     assert response.json() == expected_response_1
-    assert pwd_context.verify("secret", session.query(User).all()[0].hashed_password)
+    assert pwd_context.verify(
+        "secret", UserQuery().get_users(session)[0].hashed_password
+    )
 
-    response = client.post("users", data=registration_form_2)
+    response = client.post("/users", data=registration_form_2)
 
     assert response.status_code == status.HTTP_201_CREATED
     assert response.json() == expected_response_2
     assert pwd_context.verify(
-        "top_secret", session.query(User).all()[1].hashed_password
+        "top_secret", UserQuery().get_users(session)[1].hashed_password
     )
 
 
+def test_register_user_taken_username(client, session):
+    registration_form_1 = {
+        "username": "Mike",
+        "password": "secret",
+        "email": "mike@gmail.com",
+    }
+    registration_form_2 = {
+        "username": "Mike",
+        "password": "top_secret",
+    }
+
+    response = client.post("/users", data=registration_form_1)
+    assert response.status_code == status.HTTP_201_CREATED
+
+    response = client.post("/users", data=registration_form_2)
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+    assert len(UserQuery().get_users(session)) == 1
+
```

Comment:
I like this test. It clearly tests if we can create an user with already taken username.